pgmax/groups/enumeration.py

```diff
@@ -0,0 +1,459 @@
+"""Defines EnumerationFactorGroup and PairwiseFactorGroup."""
+
+import collections
+from dataclasses import dataclass, field
+from typing import FrozenSet, Mapping, Optional, OrderedDict, Type, Union
+
+import jax
+import jax.numpy as jnp
+import numpy as np
+
+from pgmax.factors import enumeration
+from pgmax.fg import groups, nodes
+
+
+@dataclass(frozen=True, eq=False)
+class EnumerationFactorGroup(groups.FactorGroup):
+    """Class to represent a group of EnumerationFactors.
+
+    All factors in the group are assumed to have the same set of valid configurations and
+    the same potential function. Note that the log potential function is assumed to be
+    uniform 0 unless the inheriting class includes a log_potentials argument.
+
+    Args:
+        factor_configs: Array of shape (num_val_configs, num_variables)
+            An array containing explicit enumeration of all valid configurations
```

Comment:
log_potentials should also be documented (move the documentation currently in the parent class to here)